plugins/a11y-keyboard/msd-a11y-keyboard-atspi.c

```diff
@@ -0,0 +1,193 @@
+        g_object_ref (self->priv->listener);
+#endif
+}
+
+static void
+register_deregister_events (MsdA11yKeyboardAtspi *self,
+                            gboolean              do_register)
+{
+        g_return_if_fail (MSD_IS_A11Y_KEYBOARD_ATSPI (self));
+        g_return_if_fail (ATSPI_IS_DEVICE_LISTENER (self->priv->listener));
+
+        /* register listeners for all keys with CAPS_LOCK modifier */
+        for (AtspiKeyMaskType mod_mask = 0; mod_mask < 256; mod_mask++)
+        {
+                if (! (mod_mask & (1 << ATSPI_MODIFIER_SHIFTLOCK)))
+                        continue;
+
+                if (do_register)
+                        atspi_register_keystroke_listener (self->priv->listener,
+                                                           NULL,
+                                                           mod_mask,
+                                                           1 << ATSPI_KEY_PRESSED_EVENT,
+                                                           ATSPI_KEYLISTENER_NOSYNC,
+                                                           NULL);
+                else
+                        atspi_deregister_keystroke_listener (self->priv->listener,
+                                                             NULL,
+                                                             mod_mask,
+                                                             1 << ATSPI_KEY_PRESSED_EVENT,
+                                                             NULL);
+        }
+}
+
+void
+msd_a11y_keyboard_atspi_start (MsdA11yKeyboardAtspi *self)
+{
+        g_return_if_fail (MSD_IS_A11Y_KEYBOARD_ATSPI (self));
+
+        if (self->priv->listening)
+                return;
+
+#if ! DESTROYING_ATSPI_LISTENER_CRASHES
+        /* init AT-SPI if needed */
+        atspi_init ();
+
+        self->priv->listener = atspi_device_listener_new (on_key_press_event,
+                                                          self, NULL);
+#endif
+        register_deregister_events (self, TRUE);
+        self->priv->listening = TRUE;
+}
+
+void
+msd_a11y_keyboard_atspi_stop (MsdA11yKeyboardAtspi *self)
+{
+        g_return_if_fail (MSD_IS_A11Y_KEYBOARD_ATSPI (self));
+
+        if (! self->priv->listening)
+                return;
+
+#if DESTROYING_ATSPI_LISTENER_CRASHES
+        register_deregister_events (self, FALSE);
+#else
+        g_clear_object (&self->priv->listener);
+#endif
+        self->priv->listening = FALSE;
+}
+
+#else /* ! HAVE_LIBATSPI: AT-SPI is not available, provide stubs */
+
+G_DEFINE_TYPE (MsdA11yKeyboardAtspi, msd_a11y_keyboard_atspi, G_TYPE_OBJECT)
+
+static void
+msd_a11y_keyboard_atspi_class_init (MsdA11yKeyboardAtspiClass *klass G_GNUC_UNUSED)
+{
+}
+
+static void
+msd_a11y_keyboard_atspi_init (MsdA11yKeyboardAtspi *self G_GNUC_UNUSED)
+{
+}
+
+void
+msd_a11y_keyboard_atspi_start (MsdA11yKeyboardAtspi *self G_GNUC_UNUSED)
+{
+}
+
+void
+msd_a11y_keyboard_atspi_stop (MsdA11yKeyboardAtspi *self G_GNUC_UNUSED)
+{
+}
```

Comment:
IMHO, it's not a good idea to silence the warnings.

Reply:
See https://github.com/mate-desktop/mate-settings-daemon/pull/330/files#r467228696

Reply:
This is no longer a concern as this code has been removed.